meteor/client/ui/Prompter/PrompterView.tsx

```diff
@@ -652,47 +657,147 @@ export const Prompter = translateWithTracker<PropsWithChildren<IPrompterProps>,
+			const sortedScrollAnchors = foundScrollAnchors.sort((a, b) => {
+				return a.distanceToReadPosition - b.distanceToReadPosition
+			})
+
+			// Prioritize the text anchors, then the scroll anchors:
+			return [...sortedTextAnchors, ...sortedScrollAnchors]
 		}
 
-		private restoreScrollAnchor = (scrollAnchor: ScrollAnchor) => {
-			if (scrollAnchor === null) return
-			const anchor = document.getElementById(scrollAnchor[1])
-			if (anchor) {
-				const { top } = anchor.getBoundingClientRect()
+		private restoreScrollAnchor = (scrollAnchors: ScrollAnchor[] | null) => {
+			if (scrollAnchors === null) return
+			if (!scrollAnchors.length) {
+				if (this.props.prompterData?.segments.length) {
+					// That's wierd, log the issue:
+					logger.error(`No read anchors to use after update`)
+				}
+				return
+			}
+			const readPosition = this.getReadPosition()
 
-				window.scrollBy({
-					top: top - scrollAnchor[0],
-				})
-			} else {
-				logger.error(`Read anchor could not be found after update: #${scrollAnchor[1]}`)
+			// Go through the anchors and use the first one that we find:
+			for (const scrollAnchor of scrollAnchors) {
+				const anchor = document.getElementById(scrollAnchor.anchorId)
+				if (anchor) {
+					const { top } = anchor.getBoundingClientRect()
+
+					if (scrollAnchor.offset !== null) {
+						window.scrollBy({
+							top: top - scrollAnchor.offset,
+						})
+						// We've scrolled, exit the function!
+						return
+					} else {
+						// Note: config.margin does not have to be taken into account here,
+						// the css margins magically does it for us.
+						window.scrollBy({
+							top: top - readPosition,
+						})
+						// We've scrolled, exit the function!
+						return
+					}
+				}
+			}
+			// None of the anchors where found at this point.
+
+			logger.error(
+				`Read anchor could not be found after update: ${scrollAnchors
+					.slice(0, 10)
+					.map((sa) => `"${sa.anchorId}" (${sa.offset})`)
+					.join(', ')}`
+			)
+
+			// Below is for troubleshooting, see if the anchor is in prompterData:
+			if (!this.props.prompterData) {
+				logger.error(`Read anchor troubleshooting: no prompterData`)
+				return
+			}
+
+			for (const scrollAnchor of scrollAnchors) {
+				for (const segment of this.props.prompterData.segments) {
+					if (scrollAnchor.anchorId === `segment_${segment.id}`) {
+						logger.error(`Read anchor troubleshooting: segment "${segment.id}" was found in prompterData!`)
+						return
+					}
+
+					for (const part of segment.parts) {
+						if (scrollAnchor.anchorId === `partInstance_${part.id}`) {
+							logger.error(`Read anchor troubleshooting: part "${part.id}" was found in prompterData!`)
+							return
+						}
+
+						for (const piece of part.pieces) {
+							if (scrollAnchor.anchorId === `line_${piece.id}`) {
+								logger.error(`Read anchor troubleshooting: piece "${piece.id}" was found in prompterData!`)
+								return
+							}
+						}
+					}
+				}
```

Comment:
Maybe this could be taken out to a separate method, it doesn't really have anything to do with `restoreScrollAnchor` ?